### PR TITLE
Test against PHP 8

### PR DESCRIPTION
**Pull request by alcaeus**

### CODE REVIEW

**Comment by greg0ire** [comment]
Something went very wrong, not sure what… TIL `phpunit/phpunit` depends on this package. Isn't there `COMPOSER_ROOT_VERSION` magic required for this to work?

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,6 @@ php:
 
 before_install:
   - mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || echo "xdebug not available"
-  - travis_retry composer self-update
```
Comment (greg0ire) [inline .travis.yml:13]: Could this possibly be related to the failure?

Comment (alcaeus) [reply .travis.yml:13]: Shouldn't. It's updated automatically. I'll skip this for tonight and test it tomorrow.

File: .travis.yml [hunk 2]
```diff
@@ -34,6 +36,12 @@ jobs:
         - travis_retry wget https://scrutinizer-ci.com/ocular.phar
         - travis_retry php ocular.phar code-coverage:upload --format=php-clover build/logs/clover.xml
 
+    - stage: Test
+      php: nightly
+      before_install:
+        - mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || echo "xdebug not available"
+        - composer config platform.php 7.4.99
```
Comment (greg0ire) [inline .travis.yml:43]: `composer validate --strict` should be moved to `before_install` I think, it prevents you from moving forward (in the php 8 job)

**Comment by greg0ire** [comment]
> Isn't there COMPOSER_ROOT_VERSION magic required for this to work?

Link to PR with said magic: github.com/doctrine/reflection/pull/37

**Comment by alcaeus** [comment]
`COMPOSER_ROOT_VERSION did the trick and we've now got a stable PHP 8 build (although I'd still keep `allowed_failures` for a while).

**Comment by greg0ire** [comment]
Thanks @alcaeus !

**Pull request merged**